src/compiler/scala/tools/nsc/typechecker/Contexts.scala

```diff
@@ -1950,6 +1950,11 @@ trait Contexts { self: Analyzer =>
           renamed = true
         else if (current.isWildcard && !renamed && !requireExplicit)
           result = maybeNonLocalMember(name)
+        else if (current.isGiven && !requireExplicit) {
```

Comment:
this should have the `!renamed` condition, same as the wildcard case.

`import x.{y as z, given}` means `z` should resolve to `y` not `implicit x.z` via the given selector.

Reply:
But this works in Scala 3
```
scala> object Renamer {
     |   import TX.{f as tInt, given}
     |   def test = implicitly[T[Int]]
     | }
```
so a rename is not like a mask.